File: win/src/main/csharp/net/informaticalibera/streakapp/native_/AndroidUsageBridgeImpl.cs
namespace net.informaticalibera.streakapp.native_{


public class AndroidUsageBridgeImpl : IAndroidUsageBridgeImpl {
    public bool launchPackage(String param) {
        return false;
    }

    public void openUsageAccessSettings() {
    }

    public bool isPackageLaunchable(String param) {
        return false;
    }

    public long getForegroundMillis(String param, long param1, long param2) {
        return 0;
    }

    public bool isUsageAccessGranted() {
        return false;
    }

    public bool isSupported() {
        return false;
    }

}
}

File: javase/src/main/java/net/informaticalibera/streakapp/native_/AndroidUsageBridgeImpl.java
package net.informaticalibera.streakapp.native_;

public class AndroidUsageBridgeImpl implements AndroidUsageBridge {
    public boolean launchPackage(String packageName) {
        return false;
    }

    public void openUsageAccessSettings() {
    }

    public boolean isPackageLaunchable(String packageName) {
        return true;
    }

    public long getForegroundMillis(String packageName, long fromMillis, long toMillis) {
        return 0L;
    }

    public String listLaunchableApps() {
        return "{\"apps\":["
                + "{\"name\":\"AnkiDroid\",\"package\":\"com.ichi2.anki\"},"
                + "{\"name\":\"Example Notes\",\"package\":\"com.example.notes\"},"
                + "{\"name\":\"Paper Reader\",\"package\":\"com.example.reader\"}"
                + "]}";
    }

    public boolean isUsageAccessGranted() {
        return true;
    }

    public boolean isSupported() {
        return true;
    }
}
